### ISSUE TITLE
Is it possible to use another scoring function?

### ISSUE THREAD

**Issue by KevinCrp** [gnina/gnina#190]
Hi,

Just a question
Can I add a custom scoring function (done in PyTorch) easily in gnina?

Kevin

**Comment by dkoes** [comment]
Not yet.  We use Caffe as our deep learning backend.  With a lot of effort you could figure out how to translate a convolutional pytorch model's weights into a caffe model's weights (we've done it), but I don't recommend it.

**Comment by JonasLi-19** [comment]
I wonder if I can use Caffe to re-train the CNN models, and get their new weights file, and use some command to load these new weights files to test their docking ability?  I am not sure how to resign the weights file (like .h5 format) for CNN models in GNINA...

In addition, why you team use Caffe rather than Caffe2 which is believed more convenient?

**Comment by dkoes** [comment]
Yes, the README has links to information on how to train new models.  We use Caffe because we started the project before Caffe2, tensorflow, and pytorch existed.  The plan is to move to torch in the near future.
